### PR TITLE
fix AggregateComplete tx cosign, update npm dependencies

### LINKED ISSUE #24: remove npm  warnings - update nyc dev dependency
there are npm warnings during install - updating nyc solves it, but it breaks coveralls reporting - investigation needed

### PR DESCRIPTION
**Pull request by yaaccount**
- npm dependencies has been updated to their latest versions to solve the complaints during npm install to fix #24 
- due to breaking changes with new version of instanbul/nyc, the coverage reporting has been changed to run over ts code directly (not over generated js as it was before)
- some types in generated api had to be added/fixed to make it compilable again
- AggregateComplete cosign with signatures fixed to solve #25

### CODE REVIEW

**Comment by yaaccount** [comment]
Note - the decrease in code coverage is due to newer nyc and different way how it is used (js vs. ts). The tests themselves were not changed and are still all running during npm test.

File: src/model/transaction/CosignatureTransaction.ts [hunk 1]
```diff
@@ -55,13 +56,13 @@ export class CosignatureTransaction {
      * @param gernationHash - Network generation hash
```
Comment (alvin-reyes) [inline src/model/transaction/CosignatureTransaction.ts]: change to `generationHash`

File: src/model/transaction/CosignatureTransaction.ts [hunk 2]
```diff
@@ -55,13 +56,13 @@ export class CosignatureTransaction {
...
      * @returns {CosignatureSignedTransaction}
      */
-    public static signTransactionPayload(account: Account, payload: string, gernationHash: string): CosignatureSignedTransaction {
+    public static signTransactionPayload(account: Account, payload: string, gernationHash: string, signSchema = SignSchema.SHA3): CosignatureSignedTransaction {
         /**
          * For aggregated complete transaction, cosignatories are gathered off chain announced.
          */
-        const transactionHash = VerifiableTransaction.createTransactionHash(payload, gernationHash);
+        const transactionHash = VerifiableTransaction.createTransactionHash(payload, Array.from(Convert.hexToUint8(gernationHash)));
```
Comment (alvin-reyes) [inline src/model/transaction/CosignatureTransaction.ts]: change to `generationHash`

Comment (yaaccount) [reply src/model/transaction/CosignatureTransaction.ts]: ok

File: src/model/transaction/CosignatureTransaction.ts (hunk 1 above)
Comment (yaaccount) [reply src/model/transaction/CosignatureTransaction.ts]: ok

**Pull request merged**